src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/SalesforceInputFormat.java

```diff
@@ -65,30 +48,16 @@
   private static final Gson GSON = new Gson();
   private static final Type QUERIES_TYPE = new TypeToken<List<String>>() { }.getType();
   private static final Type SCHEMAS_TYPE = new TypeToken<Map<String, String>>() { }.getType();
+  private static final Type QUERY_SPLITS_TYPE = new TypeToken<List<SalesforceSplit>>() { }.getType();
 
   @Override
   public List<InputSplit> getSplits(JobContext context) {
     Configuration configuration = context.getConfiguration();
     List<String> queries = GSON.fromJson(configuration.get(SalesforceSourceConstants.CONFIG_QUERIES), QUERIES_TYPE);
-    BulkConnection bulkConnection = getBulkConnection(configuration);
-
-    boolean enablePKChunk = configuration.getBoolean(SalesforceSourceConstants.CONFIG_PK_CHUNK_ENABLE, false);
-    if (enablePKChunk) {
-      String parent = configuration.get(SalesforceSourceConstants.CONFIG_CHUNK_PARENT);
-      int chunkSize = configuration.getInt(SalesforceSourceConstants.CONFIG_CHUNK_SIZE,
-                                           SalesforceSourceConstants.DEFAULT_PK_CHUNK_SIZE);
-
-      List<String> chunkHeaderValues = new ArrayList<>();
-      chunkHeaderValues.add(String.format(SalesforceSourceConstants.HEADER_VALUE_PK_CHUNK, chunkSize));
-      if (!Strings.isNullOrEmpty(parent)) {
-        chunkHeaderValues.add(String.format(SalesforceSourceConstants.HEADER_PK_CHUNK_PARENT, parent));
-      }
-
-      bulkConnection.addHeader(SalesforceSourceConstants.HEADER_ENABLE_PK_CHUNK, String.join(";", chunkHeaderValues));
-    }
-
+    List<SalesforceSplit> querySplits = GSON.fromJson(
+      configuration.get(SalesforceSourceConstants.CONFIG_QUERY_SPLITS), QUERY_SPLITS_TYPE);
```

Comment:
Need to handle null querySplits. Might be easier if we just passed an empty list